lib/rucio/transfertool/fts3.py

```diff
@@ -509,10 +511,10 @@ def __str__(self):
         return f'Transfer {self._transfer_id} of {self._file_metadata["scope"]}:{self._file_metadata["name"]} ' \
                f'{self._file_metadata["src_rse"]} --({self._file_metadata["request_id"]})-> {self._file_metadata["dst_rse"]}'
 
-    def initialize(self, session, logger=logging.log):
+    def initialize(self, session: "Session", logger: Callable = logging.log):
```

Comment:
Side-note: you can use the [`@abstractmethod` decorator](https://docs.python.org/3/library/abc.html#abc.abstractmethod) to mark this method as abstract, in order to enforce its implementation in child classes